components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -96,4 +96,17 @@ class Session(
     internal fun notifyObservers(block: Observer.() -> Unit) = synchronized(observers) {
         observers.forEach { it.block() }
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Session
+        if (id != other.id) return false
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return id.hashCode()
+    }
```

Comment:
Did you consider making this a data class? https://kotlinlang.org/docs/reference/data-classes.html

But I honestly have no idea how that would work with all those Observables here. Anyhow, for our use cases the UUID means equality, so 👍